src/Aspire.Hosting/DistributedApplicationRunner.cs

```diff
@@ -5,23 +5,19 @@
 using Aspire.Hosting.Publishing;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Options;
 
 namespace Aspire.Hosting;
 
-internal sealed class DistributedApplicationRunner(DistributedApplicationModel model, IOptions<PublishingOptions> options, IServiceProvider serviceProvider) : BackgroundService
+internal sealed class DistributedApplicationRunner(DistributedApplicationExecutionContext executionContext, DistributedApplicationModel model, IServiceProvider serviceProvider) : BackgroundService
 {
-    private readonly DistributedApplicationModel _model = model;
-    private readonly IOptions<PublishingOptions> _options = options;
-    private readonly IServiceProvider _serviceProvider = serviceProvider;
-
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        var publisherName = _options.Value.Publisher?.ToLowerInvariant() ?? "dcp";
-
-        var publisher = _serviceProvider.GetKeyedService<IDistributedApplicationPublisher>(publisherName)
-            ?? throw new DistributedApplicationException($"Could not find registered publisher '{publisherName}'");
+        var publisher = executionContext.Operation switch
+        {
+            DistributedApplicationOperation.Publish => serviceProvider.GetRequiredKeyedService<IDistributedApplicationPublisher>("manifest"),
+            _ => serviceProvider.GetRequiredKeyedService<IDistributedApplicationPublisher>("dcp")
+        };
```

Comment:
Undo this change. Leave the door open for publisher.

Reply:
As discussed not doing this one, but leaving `PublisherName` on the envvar callback context for 1 preview.